### PR TITLE
FEAT 60430: Add ability for wait_for_event to handle list event_id values

### LINKED ISSUE #60430: [FEATURE REQUEST] support list values for event_id in salt.wait_for_event
**Is your feature request related to a problem? Please describe.**

I am trying to use salt orchestration to wait for "old" nodes of a given role to go away because I cannot perform an operation on nodes of given role until that occurs. 

To do this, I would like to listen to `salt/presence/change` events and have `id_list` look for values in `lost`.

This currently does not work because wait_for_event expects the value for `event_id` to be a `str` and does not support `list[str]`

Offending line here: https://github.com/saltstack/salt/blob/v3003/salt/states/saltmod.py#L668

Given the following orchestration state:

```yaml
wait-for-node-death:
  salt.wait_for_event:
    - name: salt/presence/change
    - id_list:
       - some_old_node
       - another_old_node
    - timeout: 3000
    - event_id: lost
```
and an event that looks like this:

```json
{
   "data": {
      "new": [], 
      "lost": [
         "some_old_node"
      ], 
      "_stamp': '2021-06-23T15:02:04.621354"
     }, 
     "tag": "salt/presence/change"
}
```
A value exception is thrown because the linked line is doing: 

```python
["some_old_node","another_old_node"].index(["some_old_node"])
```

This is caught with: 

```
[TRACE   ] wait_for_event: Event identifier 'lost' not in id_list; skipping.
```

**Describe the solution you'd like**

When the event value retrieved from the location @ `event_id` is a list, iterate through it and pop any values that
match a value in `id_list` off of `id_list`

**Describe alternatives you've considered**

Option 1: 

Salt mine that gets updated by a reactor + orchestration when a node is lost, that way "new" nodes can tell approximately when an "old" node is lost. It seems like overkill to have a reactor + orchestration just to call salt mine.update.

Option 2: 

Use something like test.ping to try and find nodes that are no longer responding but I'd still have to do something like [`loop.until`](https://docs.saltproject.io/en/latest/ref/states/all/salt.states.loop.html) or something similar. 

If there are any other ways that this can be done, open to hearing those but I'd really like to just be able to react when a minion goes goes offline. 

I am also happy to contribute to this, it doesn't seem that difficult to accomplish, opening an FR to see if I'm missing something or if there are better/existing ways to do this.

**Additional context**

**Please Note**
If this feature request would be considered a substantial change or addition, this should go through a SEP process here https://github.com/saltstack/salt-enhancement-proposals, instead of a feature request.

### PR DESCRIPTION
**Pull request by jwyoungpm**
### What does this PR do?

Adds support for comparing `id_list` with a list when the value for `event_id` is a list in the `salt.wait_for_event` module. 

### What issues does this PR fix or reference?
Fixes #60430 

### Previous Behavior

Previously, `salt.wait_for_event` only supported `event_id` values that where `str`.

### New Behavior

In addition to the previous behavior, now supports `event_id` values that are lists. 

Given the event:

```json
{
   "data": {
      "new": [], 
      "lost": [
         "some_old_node"
      ], 
      "_stamp': '2021-06-23T15:02:04.621354"
     }, 
     "tag": "salt/presence/change"
}
```

the following `salt.wait_for_event` would not work peviously: 

```yaml
wait-for-node-death:
  salt.wait_for_event:
    - name: salt/presence/change
    - id_list:
       - some_old_node
    - timeout: 3000
    - event_id: lost
```

This PR checks if the `val` returned by the `event_id` location in the event is a list and if so, pops off any values from `id_list` that it finds in `val`

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs - **Docs do not diferentiate between value types returned by `event_id` so I did not update**
- [X] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [X] Tests written/updated

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/states/saltmod.py [hunk 1]
```diff
@@ -663,26 +663,50 @@ def wait_for_event(name, id_list, event_id="id", timeout=300, node="master"):
...
-                        log.trace(
-                            "wait_for_event: Event identifier '%s' not in "
-                            "id_list; skipping.",
-                            event_id,
-                        )
+                    if isinstance(val, list):
+
+                        val_list = [id for id in id_list if id in val]
+
+                        if len(val_list) == 0:
```
Comment (s0undt3ch) [inline salt/states/saltmod.py:670]:
```suggestion
                        if not val_list:
```

File: salt/states/saltmod.py [hunk 2]
```diff
@@ -663,26 +663,50 @@ def wait_for_event(name, id_list, event_id="id", timeout=300, node="master"):
...
+                            log.trace(
+                                "wait_for_event: Event identifier '%s' not in "
+                                "id_list; skipping",
+                                event_id,
+                            )
+                        elif len(val_list) > 0:
```
Comment (s0undt3ch) [inline salt/states/saltmod.py:676]:
```suggestion
                        elif val_list:
```

Comment (s0undt3ch) [reply salt/states/saltmod.py:676]:
```suggestion
                        else:
```
This might even be more appropriate.

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Comment by jwyoungpm** [comment]
@krionbsd sorry I haven't come back and checked on this, out on paternity :) 

Could someone have a look at this? Let me know if there are any blockers to merging.

**Review by krionbsd — APPROVED** [review]

**Comment by jwyoungpm** [comment]
@krionbsd @s0undt3ch Sorry for the very late follow up on this...I got back from paternity leave not long ago, I believe this is still a valid MR. Is this still something I could get merged? You both have approved but I can't click the merge button. 

Thanks!

**Pull request merged**